var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -414,3 +572,232 @@ def cmake_args(self):
             args.append("-DCMAKE_CXX_COMPILER_FORCED=True")
 
         return args
+
+
+class MakefileBuilder(spack.build_systems.makefile.MakefileBuilder):
+    build_directory = "src/fortran"
+
+    # Generate Compiler Specific includes
+    def edit(self, pkg, spec, prefix):
+        config = {
+            "FC": pkg.compiler.fc_names[0],
+            "FCFLAGS": "",
+            "ARCH": platform.machine(),
+            "DOCONCURRENT_FLAG": "",
+            "ARRAY_FLAG": "",
+            "OPENMP_FLAG": "",
+            "OPENACC_FLAG": "",
+            "CUDA_FLAG": "",
+            "SEQUENTIAL_FLAG": "",
+        }
+
+        # ===================================
+        #               ARM
+        # ===================================
+        if spec.compiler.name == "arm":
+            flags = "-std=f2018 " + pkg.compiler.opt_flags[4] + " -Wall -Wno-unused-variable"
+
+            if platform.machine() == "aarch64":
+                # TODO: Add MCPU option here as in -mcpu=$(MCPU)
+                flags += "-mcpu=native"
+            else:
+                # TODO: Add MARCH option here as in -march=$(MARCH)
+                flags += "-march=native"
```

Comment:
These flags are going to conflict with the flags from `archspec` that Spack will be emitting.

Is there a reason these need to be set to `native` rather than the values associated with the specific architecture targetted?

Reply:
In the new commit I tried to address this and simplify it but I am getting `error`: 

```
$ spack install babelstream%gcc@13.1.0 build_system=makefile foption=DoConcurrent target=znver3
==> Error: concretization failed for the following reasons:

   1. 'babelstream target=broadwell' cannot satisfy constraint 'target=znver3'
```

Reply:
Can you run `spack config blame packages | grep "broadwell"`? You shouldn't be stuck building for broadwell, and we definitely shouldn't say it's a broadwell and then throw non-broadwell flags.

Reply:
I think it was the issue with MACS system of Isambard, sincet then I moved to check with Phase3 system (newer)  and didn't face this issue after that